core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -388,7 +388,7 @@ private[spark] class TaskSetManager(
     if (TaskLocality.isAllowed(maxLocality, TaskLocality.NO_PREF)) {
       // Look for noPref tasks after NODE_LOCAL for minimize cross-rack traffic
       for (index <- dequeueTaskFromList(execId, pendingTasksWithNoPrefs)) {
-        return Some((index, TaskLocality.PROCESS_LOCAL, false))
+        return Some((index, TaskLocality.NO_PREF, false))
```

Comment:
hmmm....it's a typo but would not take any effect 

see discussions here:  https://github.com/apache/spark/pull/3816 (only look at the comments after Dec 30, 2014 is enough)